api/institutions/serializers.py

```diff
@@ -1,7 +1,38 @@
+import json
+import uuid
+
+from modularodm import Q
+from modularodm.exceptions import NoResultsFound
 from rest_framework import serializers as ser
+from rest_framework.exceptions import NotFound
 
+from website import security
+from website.models import Institution, User
+from framework.auth import signals
+from framework.auth.core import get_user
 from api.base.serializers import JSONAPISerializer, RelationshipField, LinksField
 
+def find_institution_by_domain(username):
+    domain = username.split('@')[1]
+    try:
+        inst = Institution.find_one(Q('domains', 'eq', domain))
+    except NoResultsFound:
+        raise NotFound
+    return inst
+
+def get_or_create_user(fullname, address):
```

Comment:
Rather than copying this, better to move `get_or_create_user` and import it in two places. I think it'd make sense to move this function to `framework.auth.core`.